geode-core/src/main/java/org/apache/geode/internal/InternalDataSerializer.java

```diff
@@ -2348,6 +2349,10 @@ public static void invokeToData(Object ds, DataOutput out) throws IOException {
    */
   public static void invokeFromData(Object ds, DataInput in)
       throws IOException, ClassNotFoundException {
+    if (ds instanceof DataSerializableFixedID) {
```

Comment:
Is there a reasonable way to eliminate this switch on type?

I see 5 instances of `instanceof DataSerializableFixedID` in our codebase.

Reply:
I'm open to suggestions on this and VersionedDataStream checks